### PR TITLE
Generified Command so that it's more easier to use when querying from…

### PR DESCRIPTION
**Pull request by shamsasari**
… LedgerTransaction

### CODE REVIEW

**Comment by joeldudleyr3** [comment]
@shamsasari - Any appetite to add a `findSingleCommand` method? We used to do this via `requireSingledCommand`. You very often want only a single type of command in a tx (e.g. only 1 type of cash command in a cash tx).

**Comment by shamsasari** [comment]
Most of the uses of `requireSingledCommand` seem to be clauses API anyway (which we're trying to move away from??). Regardless that would be a separate PR.

**Comment by mnesbit** [comment]
findCommand contains a .single already and mandates only one command meets the predicate conditions and type.

File: core/src/main/kotlin/net/corda/core/transactions/MerkleTransaction.kt [hunk 1]
```diff
@@ -29,7 +29,7 @@ interface TraversableTransaction {
     val inputs: List<StateRef>
     val attachments: List<SecureHash>
     val outputs: List<TransactionState<ContractState>>
-    val commands: List<Command>
+    val commands: List<Command<*>>
```
Comment (mnesbit) [inline core/src/main/kotlin/net/corda/core/transactions/MerkleTransaction.kt]: List<Command<CommandData>> might be friendlier to use?

**Comment by joeldudleyr3** [comment]
@mnesbit I found myself writing something like `findCommand<Move> { true }`, because I had to pass a predicate. I guess an alternative would be to have a predicate-less version?

File: core/src/main/kotlin/net/corda/core/transactions/MerkleTransaction.kt (hunk 1 above)
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/transactions/MerkleTransaction.kt]: So for both Kotlin and Java, the star-projection is still restricted to the base type:
```
 fun asd(wtx: WireTransaction) {
        val value: CommandData = wtx.commands[0].value
    }
```
Java:
```
        WireTransaction wtx;
        CommandData value = wtx.getCommands().get(0).getValue();
```

**Pull request merged**